[utils] Variant: return matching iterators for (begin|end)_(array|map)

Rechi opened the pull request:
<!--- Provide a general summary of your change in the Title above -->

## Description
[utils] Variant: return matching iterators for (begin|end)_(array|map), if type is not array or map

With this change std::distance(...) is also correct if the type isn't an array or map.
Without this change there needs to be a check for isArray respectively isObject before iterating over the elements.
<!--- Describe your change in detail -->

## Motivation and Context
Incorrect usage of CVariant at #12244.
@Montellese I don't know if you like this idea of the static array and map.
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
compiled and runtime tested on Windows and OSX
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

<!--## Screenshots (if appropriate):-->

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Montellese commented on xbmc/utils/Variant.cpp:
```diff
@@ -135,6 +135,8 @@ CVariant::CVariant()
 
 CVariant CVariant::ConstNullVariant = CVariant::VariantTypeConstNull;
+CVariant::VariantArray CVariant::EMPTY_ARRAY = VariantArray();
+CVariant::VariantMap CVariant::EMPTY_MAP = VariantMap();
```
You shouldn't need the `= VariantArray()` and `= VariantMap()` parts.

Rechi committed:
[utils] Variant: return matching iterators for (begin|end)_(array|map) if type is not array or map

With this change std::distance(...) is also correct if the type isn't an array or map.
Without this change there needs to be a check for isArray respectively isObject before iterating over the elements.

Rechi commented:
@Montellese addressed your comment and runtime tested again.

Montellese reviewed (approved):
Good to go IMO.

MartijnKaijser commented:
@Rechi could you make a backport PR?

Rechi commented:
I don't think its necessary, but I'll backport it.
